src/llmcompressor/utils/helpers.py

```diff
@@ -1091,9 +1091,11 @@ def DisableQuantization(model: torch.nn.Module):
     """
     Disable quantization from QuantizationModifier
     """
-    model.apply(disable_quantization)
-    yield
-    model.apply(enable_quantization)
+    try:
+        model.apply(disable_quantization)
+        yield
+    finally:
```

Comment:
model.apply(enable_quantization)
will always runs regardless of try being success or fail. Is this intended. 
Also if it fails I would add a log

Reply:
> model.apply(enable_quantization) will always runs regardless of try being success or fail. Is this intended.

Yes, that is exactly the intention of this change

> Also if it fails I would add a log

If an exception is raised, the user will see the raised exception

Reply:
sorry why do we want to run `model.apply(enable_quantization)` if it raises exception? Wouldn't it just exit?

And as long as the exception contains info that we can get from this function, then it should be ok. Ex. have input args useful to log causing the error.

Reply:
> sorry why do we want to run model.apply(enable_quantization) if it raises exception? Wouldn't it just exit?

For example, if the sequential pipeline fails, an exception will be thrown and then caught, in order to fall back to the layer-wise sequential pipeline. This is example of where an exception is raised but caught, and I do not want do have to think about the weird interactions that would result from this context not being properly exited.

This "try finally" pattern is, in general, good for contexts. It allows the user to not have to worry about if the context was exited properly.

Reply:
Could we move this code change to a separate PR, ideally I would like the scope of this PR to be limited to updating readme paths